lib/multicell_location/services/nrf_cloud_integration.c

```diff
@@ -60,6 +69,51 @@ const char *location_service_get_certificate_nrf_cloud(void)
 	return tls_certificate;
 }
 
+#if defined(CONFIG_NRF_CLOUD_MQTT)
+static void location_service_location_ready_cb(const struct nrf_cloud_cell_pos_result *const result)
+{
+	if (result != NULL) {
+		nrf_cloud_location.latitude = result->lat;
+		nrf_cloud_location.longitude = result->lon;
+		nrf_cloud_location.accuracy = (double)result->unc;
+	}
+	k_sem_give(&location_ready);
+}
+
+int location_service_get_cell_location_nrf_cloud(
+	const struct lte_lc_cells_info *cell_data,
+	char * const rcv_buf,
+	const size_t rcv_buf_len,
+	struct multicell_location *const location)
+{
+	ARG_UNUSED(rcv_buf);
+	ARG_UNUSED(rcv_buf_len);
+
+	int err;
+
+	k_sem_reset(&location_ready);
+
+	err = nrf_cloud_cell_pos_request(cell_data, true, location_service_location_ready_cb);
+	if (err == -EACCES) {
+		LOG_ERR("Cloud connection is not established");
+		return err;
+	} else if (err) {
+		LOG_ERR("Failed to request cell pos data, error: %d", err);
+		return err;
+	}
+
+	LOG_INF("Cellular positioning request sent");
+
+	if (k_sem_take(&location_ready, K_SECONDS(20)) == -EAGAIN) {
+		LOG_ERR("Cellular positioning data request timed out");
+		return -ETIMEDOUT;
+	}
+
+	*location = nrf_cloud_location;
+
+	return err;
+}
+#else // defined(CONFIG_NRF_CLOUD_MQTT)
```

Comment:
I thought checkpatch would only accept C-style comments, but it seems this passed. Perhaps it should be changed just to be sure CI doesn't fail down the road? @rlubos Do you know if we also accept C++ style comments now? If so, this can be ignored